go/private/context.bzl

```diff
@@ -311,6 +311,17 @@ def _library_to_source(go, attr, library, coverage_instrumented):
     if library.resolve:
         library.resolve(go, attr, source, _merge_embed)
 
+        # TODO(zbarsky): Remove this once downstream has a chance to migrate.
+        if verify_resolver_deps:
+            has_targets = False
+            for dep in source["deps"]:
+                if type(dep) != "struct":
+                    has_targets = True
+                    break
+            if has_targets:
+                print("""Detected Targets in `source["deps"]` as a result of _resolver. Please provide `GoArchive`s instead. This will be an error in the future.""")
```

Comment:
Maybe add a snippet describing how to do that (`[GoArchive]`)?

Reply:
yep, good idea